### ISSUE TITLE
Wrap Assign does not work if object has prototype.

### ISSUE THREAD

**Issue by angelvega93** [mikro-orm/mikro-orm#1083]
**Describe the bug**
Wrap Assign does not work if object has a prototype.
Object Assign works perfectly, i think that this is a bug and it should perform the same as Object Assign.

**To Reproduce**
```
@Embeddable()
export class PersonName {
  @Property()
  first: string;

  @Property({ nullable: true })
  last: string;
}

@Entity()
export class Person {
  @PrimaryKey()
  id: number;

  @Embedded({ prefix: "name_" })
  name: PersonName;
}
```
```
const person = await ctx.em.findOne(Person, { id: 1 });

const data: any = {};
data.name = new PersonName();
data.name.first = "Test";
ctx.em.assign(person, data);
await ctx.em.persistAndFlush(person);
```
It performs:
`update "person" set "name_first" = NULL where "id" = 1`

**Versions**

| Dependency | Version |
| - | - |
| node | v12.16.3 |
| typescript | v4.0.5 |
| mikro-orm | v4.3.0 |

**Comment by kodeine** [comment]
i can confirm this.

**Comment by B4nan** [comment]
Fix can be tested in latest dev version - `4.3.1-dev.9`.
